providers/azureblob/src/main/java/org/jclouds/azureblob/blobstore/AzureBlobStore.java

```diff
@@ -440,9 +440,10 @@ public String completeMultipartUpload(MultipartUpload mpu, List<MultipartPart> p
       return sync.putBlockList(mpu.containerName(), azureBlob, blocks.build());
    }
 
-   static String makeBlockId(int partNumber) {
-       return BaseEncoding.base64Url().encode(Ints.toByteArray(partNumber));
-   }
+  static String makeBlockId(int partNumber) {
+      // Azure expects a base64-encoded string ONLY. It does not support base64url encoding. 
+     return BaseEncoding.base64().encode(Ints.toByteArray(partNumber));
```

Comment:
The url encoding is handled downstream by `org.jclouds.util.Strings2`'s `urlEncode` method.